# DHI packaging: adapt nvcgo for github.com/cilium/ebpf v0.22.0.
#
# Provenance: CVE-2026-10722 is fixed in cilium/ebpf v0.22.0, which removed the
# deprecated Instruction.Sym API in favor of WithSymbol (see
# https://github.com/cilium/ebpf/releases/tag/v0.21.0). NVIDIA/libnvidia-container
# v1.20.0 still pins ebpf v0.8.0 and uses Sym in src/nvcgo/internal/cgroup/ebpf.go.
--- a/src/nvcgo/internal/cgroup/ebpf.go
+++ b/src/nvcgo/internal/cgroup/ebpf.go
@@ -159,7 +159,7 @@ func (p *program) addDevice(dev *spec.LinuxDeviceCgroup) error {
 	}
 	p.insts = append(p.insts, p.acceptBlock(dev.Allow)...)
 	// set blockSym to the first instruction we added in this iteration
-	p.insts[prevBlockLastIdx+1] = p.insts[prevBlockLastIdx+1].Sym(blockSym)
+	p.insts[prevBlockLastIdx+1] = p.insts[prevBlockLastIdx+1].WithSymbol(blockSym)
 	p.blockID++
 	return nil
 }
@@ -181,7 +181,7 @@ func (p *program) finalize(origInsts asm.Instructions, labelPrefix string) (asm.
 	// set blockSym to the first instruction of origInsts so we are able to jump to it properly
 	blockSym := fmt.Sprintf("%s-block-%d", labelPrefix, p.blockID)
 	p.insts = append(p.insts, origInsts...)
-	p.insts[lenInsts] = p.insts[lenInsts].Sym(blockSym)
+	p.insts[lenInsts] = p.insts[lenInsts].WithSymbol(blockSym)
 	p.blockID = -1
 	return p.insts, nil
 }
